_source/_posts/2022-08-19-build-crud-spring-and-vue.md

```diff
@@ -0,0 +1,1057 @@
+        config.setAllowCredentials(true);
+        // *** URL below needs to match the Vue client URL and port ***
+        config.setAllowedOrigins(Collections.singletonList("http://localhost:8080"));
+        config.setAllowedMethods(Collections.singletonList("*"));
+        config.setAllowedHeaders(Collections.singletonList("*"));
+        source.registerCorsConfiguration("/**", config);
+        FilterRegistrationBean bean = new FilterRegistrationBean<>(new CorsFilter(source));
+        bean.setOrder(Ordered.HIGHEST_PRECEDENCE);
+        return bean;
+    }
+
+    // Expose IDs of Todo items
+    @Component
+    class RestRespositoryConfigurator implements RepositoryRestConfigurer {
+        public void configureRepositoryRestConfiguration(RepositoryRestConfiguration config, CorsRegistry cors) {
+            config.exposeIdsFor(Todo.class);
+        }
+    }
+    
+}
+
+```
+
+This demo application does three things that are helpful for demonstration purposes. First, it loads some test todo items into the repository. 
+
+Second, it configures the REST repository to expose IDs for the todo items. 
+
+Third, it defines a filter to allow cross-origin requests from `http://localhost:8080`. This is necessary so that the Vue application, which is loaded from `http://localhost:9000` via the local test server, can load data from the Spring Boot resource server at `http://localhost:8080`. 
+
+For more info on CORS (cross-origin resource sharing), take a look at [the Mozilla docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS).
+
+Now, create the data model for the todo items. 
+
+`src/main/java/com/example/demo/Todo.java`
+
+```java
+package com.example.demo;
+
+import lombok.*;
+
+import javax.persistence.Id;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Entity;
+
+@Entity
+@Data
+@NoArgsConstructor
+public class Todo {
+
+    @Id @GeneratedValue
+    private Long id;
+
+    @NonNull
+    private String title;
+
+    private Boolean completed = false;
+
+}
+```
+
+Notice the use of the Lombok annotations (`@Entity`, `@Data`, and `@NoArgsConstructor`) to keep the code simple and clean.
+
+The todo items have two fields: a title string and a completed boolean. The fields are annotated with Spring Data JPA annotations that allow the Java class to be mapped to a database table for persistence.
+
+Create a repository to persist the data model.
+
+`src/main/java/com/example/demo/TodoRepository.java`
+
+```java
+package com.example.demo;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.rest.core.annotation.RepositoryRestResource;
+
+@RepositoryRestResource
+interface TodoRepository extends JpaRepository<Todo, Long> {}
+```
+
+This is a Spring JpaRepository that can persist the data model you just defined. Because it is annotated with `@RepositoryRestResource` (and because the `data-rest` dependency was included), this repository will be automatically exposed as a web resource.
```

Comment:
```suggestion
This is a Spring Data JpaRepository that can persist the data model you just defined. Because it is annotated with `@RepositoryRestResource` (and because the `data-rest` dependency was included), this repository will be automatically exposed as a web resource.
```